geode-core/src/main/java/org/apache/geode/cache/query/internal/index/IndexElemArray.java

```diff
@@ -243,7 +243,7 @@ public boolean addAll(Collection c) {
     synchronized (lock) {
       ensureCapacity(size + numNew);
       System.arraycopy(a, 0, elementData, size, numNew);
-      size += numNew;
+      size += (byte) (numNew & 0xFF);
```

Comment:
it seems like addAll is still broken. Since "size" is a byte does line 244: ensureCapacity(size+numNew) work? If so it seems like it would allow you to grow elementData beyond the max of 127 (max signed byte) and then your new code make sure that we do not inc size by more than 127 but that can still overflow and now size will be in the range -128..127 even though elementData.length might be more than 127. It seems like ensureCapacity should be make sure we do not grow elementData beyond the max of 127 since "size" is a byte. If all this is correct I don't think you are doing any harm here; we just need a bug on this class.